#!/usr/bin/env Rscript

library(facets)

args = commandArgs(trailingOnly=TRUE)

d   = args[1]
sid = args[2]
wdir = "/data/NHLBI_BCB/Mehdi/SEQC-WG1/22-CNV-MS/08_FACET/"



setwd(paste(wdir,d,"/",sid,"/",sep=""))

# Need to run snp-pileup script (link below) to generate input for facets for each normal-tumor sample pair
# https://github.com/mskcc/facets/tree/master/inst/extcode
# snp-pileup -g -q15 -Q20 -P100 -r25,0 vcffile outputfile normalbam tumorbam
# /data/CCBR_Pipeliner/db/PipeDB/lib/Homo_sapiens_assembly38.dbsnp138.vcf.gz

set.seed(1234)

# datafile = system.file("extdata", "stomach.csv.gz", package="facets") # read SNP data file from above, as input
# rcmat = readSnpMatrix(datafile) # Reads a snp read count matrix generated by the snp-pileup code included and prepares the read counts data frame needed for ‘preProcSample’
# xx = preProcSample(rcmat) # Processes a snp read count matrix and generates a segmentation tree

rcmat = readSnpMatrix(filename=paste(wdir,d,"/",sid,"/",sid,".gz",sep=""))
dim(rcmat)

#25, 75, 150
cval1=75

#250, 1000, 10000
densewindowsize=1000
xx = preProcSample(rcmat, gbuild="hg38", cval=cval1, snp.nbhd=densewindowsize)

#150, 300, 450
cval2=300

oo=procSample(xx,cval=cval2) # Processes the output from preProcSample for given cval and min.nhet
oo$dipLogR # We call the logR for the 2-copy state logR0 which is output below
fit=emcncf(oo) # Uses genotype mixture model to estimate the cluster specific copy number and cellular fraction. Uses estimates based on the cnlr.median and mafR as initial values for the EM iteration
head(fit$cncf) # The segmentation result and the EM fit output looks like this
write.table(fit$cncf, file=paste(wdir,d,"/",sid,"/",sid,".",cval1,".",cval2,".",densewindowsize,".facets.segmentation.out.txt",sep=""))

# Estimated tumor sample purity and ploidy are reported
fit$purity
fit$ploidy
pdf(paste(wdir,d,"/",sid,"/",sid,".",cval1,".",cval2,".",densewindowsize,".facets.genome.wide.profile.pdf",sep=""))
plotSample(x=oo,emfit=fit) # We provide a plot function to visualize the genome-wide profile and FACETS output
# logRlogORspider(oo$out, oo$dipLogR) # A diagnostic plot for the fit can be generated using the function
dev.off()

fit2=emcncf2(oo) # for any subclonal clusters identified in the tumor sample
head(fit2$cncf) # output file

write.table(fit2$cncf, file=paste(wdir,d,"/",sid,"/",sid,".",cval1,".",cval2,".",densewindowsize,".facets.segmentation.clustering.out.txt",sep=""))

